src/Ocelot/Request/Middleware/DownstreamRequest.cs

```diff
@@ -36,7 +37,11 @@ public DownstreamRequest(HttpRequestMessage request)
 
         public string Query { get; set; }
 
-        public HttpContent Content { get; set; }
+        public virtual bool HasContent { get => _request?.Content != null; }
+
+        public virtual Task<string> ReadContentAsync() => HasContent
```

Comment:
I think we shouldn't allow that kind of virtual methods for the purpose of mocks in unit tests. It's virtual so it could be modified and it would hide the original implementation.

Reply:
Let's double check this after release!
Now I'm going to revert #1172 ...